From: Frank Haverkamp <haver@linux.vnet.ibm.com>
Date: Wed, 10 Sep 2014 16:37:49 +0200
Subject: [PATCH] GenWQE: Do not modify return code of
 genwqe_set_interrupt_capability
Patch-mainline: v3.18-rc1
Git-commit: 2d880ccfa9be92a10ea19f5a8f7e4be2a7d45e4d
References: bsc#949731,LTC#131502

Follow up patch to the one from Sebastian Ott. There is no need to
change the return code once it fails. And Sebastians version is tested
now and works nicely on our test-system.

Signed-off-by: Frank Haverkamp <haver@linux.vnet.ibm.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/misc/genwqe/card_ddcb.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/misc/genwqe/card_ddcb.c
+++ b/drivers/misc/genwqe/card_ddcb.c
@@ -1252,10 +1252,8 @@ int genwqe_setup_service_layer(struct ge
 	}
 
 	rc = genwqe_set_interrupt_capability(cd, GENWQE_MSI_IRQS);
-	if (rc) {
-		rc = -ENODEV;
+	if (rc)
 		goto stop_kthread;
-	}
 
 	/*
 	 * We must have all wait-queues initialized when we enable the
